pkg/controller/service/controller.go

```diff
@@ -133,16 +136,18 @@ func New(
 	}
 
 	s := &Controller{
-		cloud:            cloud,
-		knownHosts:       []*v1.Node{},
-		kubeClient:       kubeClient,
-		clusterName:      clusterName,
-		cache:            &serviceCache{serviceMap: make(map[string]*cachedService)},
-		eventBroadcaster: broadcaster,
-		eventRecorder:    recorder,
-		nodeLister:       nodeInformer.Lister(),
-		nodeListerSynced: nodeInformer.Informer().HasSynced,
-		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "service"),
+		cloud:                              cloud,
+		knownHosts:                         []*v1.Node{},
+		kubeClient:                         kubeClient,
+		clusterName:                        clusterName,
+		cache:                              &serviceCache{serviceMap: make(map[string]*cachedService)},
+		eventBroadcaster:                   broadcaster,
+		eventRecorder:                      recorder,
+		nodeLister:                         nodeInformer.Lister(),
+		nodeListerSynced:                   nodeInformer.Informer().HasSynced,
+		queue:                              workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "service"),
+		legacyNodeRoleFeatureEnabled:       utilfeature.DefaultFeatureGate.Enabled(legacyNodeRoleBehaviorFeature),
```

Comment:
pass the `k8s.io/component-base/featuregate.FeatureGate` to this constructor so that this package can remove the `utilfeature` dependency.

Reply:
Good point, updated